bedwars-plugin/src/main/java/com/tomkeuper/bedwars/commands/bedwars/subcmds/regular/CmdLeave.java

```diff
@@ -53,12 +59,37 @@ public boolean execute(String[] args, CommandSender s) {
         if (s instanceof ConsoleCommandSender) return false;
         Player p = (Player) s;
 
-        if (cancel(p.getUniqueId())) return true;
-        update(p.getUniqueId());
-        IArena a = Arena.getArenaByPlayer(p);
-
-        Misc.moveToLobbyOrKick(p, a, a != null && a.isSpectator(p.getUniqueId()));
-        return true;
+        if (p.getWorld().getName().equalsIgnoreCase(BedWars.getLobbyWorld())) {
+            if (cancel(p.getUniqueId())) return true;
+            update(p.getUniqueId());
+            IArena a = Arena.getArenaByPlayer(p);
+            Misc.moveToLobbyOrKick(p, a, a != null && a.isSpectator(p.getUniqueId()));
+            return true;
+        } else {
+            if (cancel(p.getUniqueId())) return true;
```

Comment:
cant this be pulled out of the if statement?

Reply:
> cant this be pulled out of the if statement?

which code?

Reply:
the `if (cancel(p.getUniqueId())) return true;`

Reply:
> cant this be pulled out of the if statement?

This is the original code

Reply:
what do you mean with orignial code?

Reply:
> what do you mean with orignial code?

do i really need to change it? I don't see a need to change

Reply:
I'm asking you if it can, and to me it looks like it can.
But what do you mean with original code?

Reply:
> I'm asking you if it can, and to me it looks like it can. But what do you mean with original code?

Original code means previously written code.

Reply:
And where did you get this code from?

Reply:
> And where did you get this code from?

Are you referring to the code written in this pull request?

Reply:
> > And where did you get this code from?
> 
> Are you referring to the code written in this pull request?

yes, since you are talking about 'the original' code

Reply:
> > > And where did you get this code from?
> > 
> > 
> > Are you referring to the code written in this pull request?
> 
> yes, since you are talking about 'the original' code

The leave delay code was written by me

Reply:
Okay, getting back to the original question, the if statement is a duplicate and it looks to me like it can be extracted, if thats the case then it also should be extracted.

Reply:
> Okay, getting back to the original question, the if statement is a duplicate and it looks to me like it can be extracted, if thats the case then it also should be extracted.

The first if statement only works in the lobby world. The second if statement case only works in arena. So I wrote this.

Reply:
![image](https://github.com/tomkeuper/BedWars2023/assets/29728836/2786c49c-80fb-43aa-9c57-6b20e3f4219b)
these?

Reply:
> ![image](https://user-images.githubusercontent.com/29728836/238133898-2786c49c-80fb-43aa-9c57-6b20e3f4219b.png) these?

It seems like a command use delay, can I remove it?

Reply:
I dont understand what you are saying. You wrote this code right?

Reply:
> I dont understand what you are saying. You wrote this code right?

The part you marked was coded by andrei1058. The reason I added this code again is probably because of the command delay.

Reply:
okay so to get things straight, you did or did not write the code in this PR? and then im talking about the original code. 

Because if you are saying that the code came from andrei I kinda want to know where from.

Reply:
> okay so to get things straight, you did or did not write the code in this PR? and then im talking about the original code.
> 
> Because if you are saying that the code came from andrei I kinda want to know where from.

The cooldown code was originally there.
https://github.com/andrei1058/BedWars1058/blob/master/bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/regular/CmdLeave.java

Reply:
> There is no cooldown code there. just the code to leave the arena. But I guess thats fine. The code I highlighted before just checks if the event is canceled.
> 
> But looking at the code you provided you have an if statement that checks if the player is in a lobby world or not so no matter what the outcome is, it will still run partiaclly the same code. (which is a waste of space and resources.) You should always extract the duplicate lines to outside the if statement. And it also looks like you can check if the player is in an arena or not directly within the if statement of checking whether the player is in a lobby or not.

i removed it

Reply:
Thats not what im saying you should do. extracting code does not mean removing. If you want I can implement this aswell so you can see what i mean

Reply:
> Thats not what im saying you should do. extracting code does not mean removing. If you want I can implement this aswell so you can see what i mean

This code is just this. It is also used elsewhere. There is a cooldown on opening the stats gui. So it seems to be used here too.